### PR TITLE
sdk: close all positions

### PR DESCRIPTION
**Pull request by crispheaney**
Fixes DPE-147

### CODE REVIEW

File: sdk/src/clearingHouse.ts [hunk 1]
```diff
@@ -1160,6 +1161,31 @@ export class ClearingHouse {
 		);
 	}
 
+	public async closeAllPositions(
+		userPositionsAccount: UserPositionsAccount,
+		discountToken?: PublicKey,
+		referrer?: PublicKey
+	): Promise<TransactionSignature> {
+		const ixs: TransactionInstruction[] = [];
+		for (const userPosition of userPositionsAccount.positions) {
```
[the remark this answers is not in the document]
Comment (0xbigz) [reply sdk/src/clearingHouse.ts:1170]: think the goal is to make it an atomic "all or nothing"

[the remark this answers is not in the document]
Comment (0xbigz) [reply sdk/src/clearingHouse.ts:1170]: do you mean just make the for loop parallel to add the ixs to the transaction?

[the remark this answers is not in the document]
Comment (lowkeynicc) [reply sdk/src/clearingHouse.ts:1170]: yea, just a nit (assuming order of ixs doesn't matter)

[the remark this answers is not in the document]
Comment (crispheaney) [reply sdk/src/clearingHouse.ts:1170]: all the await statements should be cached after the first call, so in practice i dont think making it in parallel will speed it up. Good catch though!

**Pull request merged**